pkg/sql/schemachanger/scplan/internal/scstage/build.go

```diff
@@ -316,23 +422,33 @@ func (sb stageBuilder) isOutgoingOpEdgeAllowed(e *scgraph.OpEdge) bool {
 	if e.Type() != sb.opType {
 		return false
 	}
-	// We allow non-revertible ops to be included at stages preceding
-	// PostCommitNonRevertible if nothing left in the schema change at this
-	// point can fail. The caller is responsible for detecting whether any
-	// non-revertible operations are included in a phase before
-	// PostCommitNonRevertible and adjusting the phase accordingly. This is
-	// critical to allow op-edges which might otherwise be revertible to be
-	// grouped with non-revertible operations.
-	if !sb.bc.isRevertibilityIgnored &&
-		sb.bs.phase < scop.PostCommitNonRevertiblePhase &&
-		!e.Revertible() &&
-		// We can't act on the knowledge that nothing remaining can fail while in
-		// StatementPhase because we don't know about what future targets may
-		// show up which could fail.
-		(sb.bs.phase < scop.PreCommitPhase || sb.anyRemainingOpsCanFail) {
-		return false
+	// At this point, consider whether this edge is allowed based on the current
+	// phase, whether the edge is revertible, and other information.
+	switch sb.bs.currentPhase {
+	case scop.StatementPhase:
+		// We ignore revertibility in the statement phase. This ensures that
+		// the side effects of a schema change become immediately visible
+		// to the transaction. For example, dropping a table in an explicit
+		// transaction should make it impossible to query that table later
+		// in the transaction.
+		// Regardless, we cannot allow the element to transition more than once
+		// within the statement transaction.
+		return scpb.AsTargetStatus(e.From().CurrentStatus) != scpb.InvalidTarget
```

Comment:
I've read this a few times and I still don't get how the logic of this line corresponds to the comment. What is special about `InvalidTarget` here? Can you spell it out in more detail for future readers?

Reply:
Will do! It's checking that the op-edge originates from the source node in the opgen path.If that's not true then we'd be performing more than one transition in the transaction.